src/tlo/methods/causes.py

```diff
@@ -67,8 +68,9 @@ def check_cause(_cause: Cause, _acceptable_causes: set):
                         f"Conflict in declared cause {tlo_cause} by {m.name}. " \
                         f"A different specification has already been registered."
 
-                # If ok, update these causes to the master dict of all causes of death
-                collected_causes.update({tlo_cause: cause})
+                # If ok, update these causes to the master dict of all causes of death  (copies to prevent any
+                # accidental editing of the modules' declarations).
+                collected_causes.update({copy(tlo_cause): copy(cause)})
```

Comment:
What data types are `tlo_cause` and `cause` here? Most built-in mutable container types (`list`, `dict`, `set`) have a `copy` method that would avoid the need to import `copy` function. If `tlo_cause` is an immutable type (for example `str` or `tuple`), which is likely as generally `dict` keys need to be immutable to be hashable, then no need to create a copy as underlying container cannot be updated after creation. 

Also tangential to changes here, for adding a single entry to a dictionary something like `collected_causes[tlo_cause] = cause.copy()` would I think be both clearer and more performant than using `update` method (which is more meant for merging in a dictionary with multiple items).

Reply:
I can make this:
```python
collected_causes[tlo_cause] = copy(cause)
```

`tlo_cause` is a string, but `cause` is a custom type I've written

Reply:
Yep if `tlo_cause` is a string then no need to copy (as it's immutable so cannot be modified in place) and using `copy` function on `cause` makes sense if a custom type.